### PR TITLE
fix: sampled profile format validation errors

### PR DESCRIPTION
**Pull request by armcknight**
This was causing profiles with the new schema to fail validation.

#skip-changelog

### CODE REVIEW

File: Sources/Sentry/SentryProfiler.mm [hunk 1]
```diff
@@ -522,6 +522,9 @@ - (void)captureEnvelope
     const auto profileDuration = getDurationNs(_startTimestamp, _endTimestamp);
     profile[@"duration_ns"] = [@(profileDuration) stringValue];
     profile[@"truncation_reason"] = profilerTruncationReasonName(_truncationReason);
+    profile[@"platform"] = _transactions.firstObject.platform;
+    profile[@"environment"] = _hub.scope.environmentString ?: _hub.getClient.options.environment ?: kSentryDefaultEnvironment;
+    profile[@"timestamp"] = [[SentryCurrentDate date] sentry_toIso8601String];
```
Comment (armcknight) [inline Sources/Sentry/SentryProfiler.mm:527]: Please confirm that a string such as `2022-11-02T21:16:20.877Z` would validate? @phacops

File: Sources/Sentry/SentryProfiler.mm [hunk 2]
```diff
@@ -585,12 +588,15 @@ - (void)captureEnvelope
                           : (unsigned long long)(
                               [transaction.timestamp timeIntervalSinceDate:_startDate] * 1e9)];
         [transactionsInfo addObject:@{
-            @"environment" : _hub.scope.environmentString ?: _hub.getClient.options.environment ?: kSentryDefaultEnvironment,
             @"id" : transaction.eventId.sentryIdString,
             @"trace_id" : transaction.trace.context.traceId.sentryIdString,
             @"name" : transaction.transaction,
             @"relative_start_ns" : relativeStart,
-            @"relative_end_ns" : relativeEnd
+            @"relative_end_ns" : relativeEnd,
```
Comment (Zylphrex) [inline Sources/Sentry/SentryProfiler.mm:595]: Do we have a TODO somewhere to make sure the `relative_end_ns` is correct?

**Review by Zylphrex — APPROVED** [review]
LGTM pending the follow-ups for the `relative_end_ns` and `active_thread_id` 👍

File: Sources/Sentry/SentryProfiler.mm (hunk 2 above)
Comment (armcknight) [reply Sources/Sentry/SentryProfiler.mm:595]: I just filed an issue here for this, thanks for reminding me! https://github.com/getsentry/sentry-cocoa/issues/2352

File: Sources/Sentry/SentryProfiler.mm (hunk 1 above)
Comment (phacops) [reply Sources/Sentry/SentryProfiler.mm:527]: It would.

**Review by phacops — APPROVED** [review]

File: Sources/Sentry/SentryProfiler.mm [hunk 3]
```diff
@@ -585,12 +588,15 @@ - (void)captureEnvelope
...
+            @"active_thread_id" :
+                mainThreadID // TODO: we are just using the main thread ID for all transactions to
```
Comment (armcknight) [inline Sources/Sentry/SentryProfiler.mm:597]: TODO captures in https://github.com/getsentry/sentry-cocoa/issues/2356

**Pull request merged**